server/src/test/java/io/crate/expression/operator/EqOperatorTest.java

```diff
@@ -96,4 +110,32 @@ public void testEvaluateEqOperator() {
         assertNormalize("1.2 = null", isLiteral(null));
         assertNormalize("'foo' = null", isLiteral(null));
     }
+    @Test
+    public void test_array_equals_empty_array_on_all_types() throws Exception {
+        List<DataType<?>> allTypes = new ArrayList<>();
+        DataTypeTesting.ALL_STORED_TYPES_EXCEPT_ARRAYS.forEach(type -> {
+            allTypes.add(type);
```

Comment:
as discussed, was going to add nested array check but storage of 2d is not supported